### PR TITLE
Refine imports in `Reflection.AST`

### LINKED ISSUE #1864: Dependencies of `Reflection.AST.Argument.Information`
Type-checking the above module seems to drag in a huge amount of unnecessary stuff. It must be possible to simplify the dependencies. Also check other `Reflection.AST` modules!

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Works towards #1864

### CODE REVIEW

File: src/Reflection/AST/Name.agda [hunk 1]
```diff
@@ -8,11 +8,11 @@
 
 module Reflection.AST.Name where
 
-open import Data.List.Base              using (List)
+open import Data.List.Base               using (List)
 import Data.Product.Properties as Prodₚ using (≡-dec)
 import Data.Word.Properties as Wₚ       using (_≟_)
-open import Function.Base               using (_on_)
-open import Relation.Nullary.Decidable                 using (map′)
+open import Function.Base                using (_on_)
```
Comment (Saransh-cpp) [inline src/Reflection/AST/Name.agda:14]:
```suggestion
open import Function.Base               using (_on_)
```

File: src/Reflection/AST/Name.agda [hunk 2]
```diff
@@ -8,11 +8,11 @@
 
 module Reflection.AST.Name where
 
-open import Data.List.Base              using (List)
+open import Data.List.Base               using (List)
```
Comment (Saransh-cpp) [inline src/Reflection/AST/Name.agda:11]:
```suggestion
open import Data.List.Base              using (List)
```

**Comment by jamesmckinna** [comment]
See also the discussion on #1876 (which did a first cut on the tidying up the imports), where I drew attention to the opportunity for additional streamlining of this subhierarchy by cleaning up the decidability proofs. Any chance of looking at those, too?

**Comment by MatthewDaggitt** [comment]
@Saransh-cpp any chance you can unmark this as a draft so I can merge this in and then you can make any changes following on from @jamesmckinna's suggestions in a separate PR?

**Comment by Saransh-cpp** [comment]
Yes! This PR will automatically close the corresponding issue, so you might have to reopen that/unlink it from this PR manually.

**Pull request merged**